### PR TITLE
test/e2e/karmadactl_test.go: test describe command

### PR DESCRIPTION
**Pull request by mohamedawnallah**
**Description**

In this commit, we test `karmadactl describe` command on existing resource, non-existing resource, non-existing namespace, and invalid resource type.

**What type of PR is this?**

/kind failing-test

**Which issue(s) this PR fixes**:

Part of #4544

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by Vacant2333 — APPROVED** [review]
thanks! this looks good to me~
/approve

File: test/e2e/karmadactl_test.go [hunk 1]
```diff
@@ -610,3 +610,108 @@ var _ = ginkgo.Describe("Karmadactl top testing", ginkgo.Labels{NeedCreateCluste
...
+			namespace = fmt.Sprintf("karmadatest-%s", rand.String(6))
+			podName = fmt.Sprintf("test-pod-%s", rand.String(6))
+			pod = helper.NewPod(namespace, podName)
+
+			// Create the namespace and pod in the member cluster.
+			ns := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: namespace,
+				},
+			}
```
Comment (zhzhuang-zju) [inline test/e2e/karmadactl_test.go:637]:
```suggestion
			namespace = fmt.Sprintf("karmadatest-%s", rand.String(RandomStrLength))
			podName = podNamePrefix + rand.String(RandomStrLength)
			pod = helper.NewPod(namespace, podName)
			ns =  helper.NewNamespace(namespace)
			
			// Create the namespace and pod in the member cluster.
			
```

Comment (zhzhuang-zju) [reply test/e2e/karmadactl_test.go:637]: I'll omit the rest of the same review

**Review by zhzhuang-zju — COMMENTED** [review]
others LGTM

File: test/e2e/karmadactl_test.go (hunk 1 above)
Comment (mohamedawnallah) [reply test/e2e/karmadactl_test.go]: Thanks! Next time, I will check for any global constants before using any magic numbers. 👍

Comment (mohamedawnallah) [reply test/e2e/karmadactl_test.go]: Updated it!

Comment (mohamedawnallah) [reply test/e2e/karmadactl_test.go]: Ah noticed now using a public function for constructing namespace instead of row construction 👍

Comment (mohamedawnallah) [reply test/e2e/karmadactl_test.go]: Update it now!

**Comment by mohamedawnallah** [comment]
I will ask for review when GitHub CI tests finish 👍

**Comment by zhzhuang-zju** [comment]
/lgtm
thanks~ cc @Vacant2333 @XiShanYongYe-Chang

**Comment by mohamedawnallah** [comment]
This also needs rebase but let's merge those sequentially. Please ping me once it is time for merge so I rebase :)

**Comment by mohamedawnallah** [comment]
Rebased now. Will ask for review when the GitHub CI tests finish 👍

**Comment by mohamedawnallah** [comment]
@XiShanYongYe-Chang The GitHub CI tests are all finished and passed. I think it is ready for merge :)

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~
/approve

**Pull request merged**